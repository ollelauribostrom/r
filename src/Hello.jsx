// @flow

import React from 'react';

type HelloProps = {
  name: string,
};

export const Hello = ({ name }: HelloProps) => <span>Hello {name}!</span>;
