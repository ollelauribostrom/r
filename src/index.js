// @flow

export { Hello } from './Hello';
