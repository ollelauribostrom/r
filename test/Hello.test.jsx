import React from 'react';
import { shallow } from 'enzyme';
import { Hello } from '../src';

describe('Tests for Hello component', () => {
  it('renders the correct text', () => {
    const wrapper = shallow(<Hello name="test" />);
    expect(wrapper.text()).toEqual('Hello test!');
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<Hello name="test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
