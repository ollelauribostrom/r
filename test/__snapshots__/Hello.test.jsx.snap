// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Hello component matches snapshot 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Hello 
  test
  !
</button>
`;
